gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql

```diff
@@ -0,0 +1,111 @@
+/* gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql */
+
+/*
+ This is part of arenadata_toolkit API for ADB Bundle.
+ This function collects information in db_files_current and db_files_history tables.
+ */
+CREATE OR REPLACE FUNCTION arenadata_toolkit.adb_collect_table_stats()
+RETURNS VOID
+AS $$
+BEGIN
+	IF NOT EXISTS (
+		SELECT
+		FROM pg_partition pp
+		JOIN pg_partition_rule pr ON pr.paroid = pp.oid
+		WHERE
+			pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass
+			AND
+			CASE
+				WHEN pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass
+				THEN now() BETWEEN
+					CAST(substring(pg_get_expr(pr.parrangestart, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
+						AS TIMESTAMP WITHOUT TIME ZONE)
+					AND
+					CAST(substring(pg_get_expr(pr.parrangeend, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
+						AS TIMESTAMP WITHOUT TIME ZONE)
+				ELSE FALSE
+			END)
+	THEN
+		EXECUTE FORMAT($fmt$ALTER TABLE arenadata_toolkit.db_files_history SPLIT DEFAULT PARTITION
+			START (date %1$L) INCLUSIVE
+			END (date %2$L) EXCLUSIVE

```

Comment:
BETWEEN is used in the SELECT above, so parrangeend is included, but end date is excluded here

Reply:
Updated